inttest/pvc/pvc_test.go

```diff
@@ -0,0 +1,219 @@
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+
+	km "github.com/k0sproject/k0smotron/api/k0smotron.io/v1beta1"
+	"github.com/k0sproject/k0smotron/inttest/util"
+)
+
+type PVCSuite struct {
+	common.FootlooseSuite
+}
+
+func (s *PVCSuite) TestK0sGetsUp() {
+	s.T().Log("starting k0s")
+	s.Require().NoError(s.InitController(0, "--disable-components=konnectivity-server,metrics-server"))
+	s.Require().NoError(s.RunWorkers())
+
+	kc, err := s.KubeClient(s.ControllerNode(0))
+	s.Require().NoError(err)
+	rc, err := s.GetKubeConfig(s.ControllerNode(0))
+	s.Require().NoError(err)
+
+	err = s.WaitForNodeReady(s.WorkerNode(0), kc)
+	s.NoError(err)
+
+	// create folder for k0smotron persistent volume
+	ssh, err := s.SSH(s.Context(), s.ControllerNode(0))
+	s.Require().NoError(err)
+	defer ssh.Disconnect()
+
+	s.Require().NoError(s.ImportK0smotronImages(s.Context()))
+
+	s.T().Log("deploying k0smotron operator")
+	s.Require().NoError(util.InstallK0smotronOperator(s.Context(), kc, rc))
+	s.Require().NoError(common.WaitForDeployment(s.Context(), kc, "k0smotron-controller-manager", "k0smotron"))
+
+	s.T().Log("waiting for seaweed storage to be ready")
+	s.Require().NoError(common.WaitForStatefulSet(s.Context(), kc, "seaweedfs-volume", "seaweedfs-operator-system"))
+	s.Require().NoError(common.WaitForDeployment(s.Context(), kc, "seaweedfs-csi-driver-controller", "seaweedfs-operator-system"))
+
+	s.T().Log("deploying k0smotron cluster")
+	s.createK0smotronCluster(s.Context(), kc)
+	s.Require().NoError(common.WaitForStatefulSet(s.Context(), kc, "kmc-kmc-test", "kmc-test"))
+
+	s.T().Log("Starting portforward")
+	fw, err := util.GetPortForwarder(rc, "kmc-kmc-test-0", "kmc-test", 6443)
+	s.Require().NoError(err)
+
+	go fw.Start(s.Require().NoError)
+	defer fw.Close()
+
+	<-fw.ReadyChan
+
+	localPort, err := fw.LocalPort()
+	s.Require().NoError(err)
+	kmcKC, err := util.GetKMCClientSet(s.Context(), kc, "kmc-test", "kmc-test", localPort)
+	s.Require().NoError(err)
+
+	_, err = kmcKC.CoreV1().ConfigMaps("default").Create(s.Context(), &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-config",
+		},
+		Data: map[string]string{
+			"foo": "bar",
+		},
+	}, metav1.CreateOptions{})
+	s.Require().NoError(err)
+
+	_, err = ssh.ExecWithOutput(s.Context(), fmt.Sprintf("%s kc exec -n kmc-test kmc-kmc-test-0 -- touch /var/lib/k0s/pvc-test", s.K0sFullPath))
+	s.Require().NoError(err)
+
+	s.T().Log("updating k0smotron cluster")
+	s.updateK0smotronCluster(s.Context(), rc)
+
+	err = wait.PollUntilContextCancel(s.Context(), time.Second, true, func(_ context.Context) (done bool, err error) {
+		sts, err := kc.AppsV1().StatefulSets("kmc-test").Get(s.Context(), "kmc-kmc-test", metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+
+		return "250Mi" == sts.Spec.VolumeClaimTemplates[0].Spec.Resources.Requests.Storage().String(), nil
```

Comment:
I think we should also verify that the etcd volume has been resized to 70Mi

Reply:
Yeah, makes a lot of sense, thanks! Added the check for etcd statefulset